packages/rocketchat-contacts/server/index.js

```diff
@@ -0,0 +1,85 @@
+/* globals SyncedCron */
+
+import { Meteor } from 'meteor/meteor';
+const service = require('./service.js');
+const provider = new service.Provider();
+
+function refreshContactsHashMap() {
+	let phoneFieldName = '';
+
+	RocketChat.settings.get('Contacts_Phone_Custom_Field_Name', function(name, fieldName) {
+		phoneFieldName = fieldName;
+	});
+
+	const contacts = [];
+	const cursor = Meteor.users.find({ active:true });
+
+	if (phoneFieldName) {
+		const phoneFieldArray = phoneFieldName.split(',');
+
+		cursor.forEach((user) => {
+			let dict = user;
+			const discoverable = RocketChat.getUserPreference(user, 'isPublicAccount');
+			if (discoverable !== false) {
+				for (let i = 0;i < phoneFieldArray.length - 1;i++) {
+					if (phoneFieldArray[i] in dict) {
+						dict = dict[phoneFieldArray[i]];
+					}
+				}
+				const phone = dict[phoneFieldArray[phoneFieldArray.length - 1]];
+
+				if (phone) {
+					contacts.push({ d:phone, u:user.username });
+				}
+
+				if ('emails' in user) {
+					user.emails.forEach((email) => {
+						if (email.verified) {
+							contacts.push({ d:email.address, u:user.username });
+						}
+					});
+				}
+			}
+		});
+	} else {
+		cursor.forEach((user) => {
+			const discoverable = RocketChat.getUserPreference(user, 'isPublicAccount');
+			if (discoverable !== false && 'emails' in user) {
```

Comment:
same comment.....  there will be no "emails" but only "services,viasatsso,email"